src/main/java/com/microsoft/azure/functions/annotation/CosmosDBOutput.java

```diff
@@ -12,66 +12,116 @@
  * public String cosmosDbAddItem(
  *    {@literal @}HttpTrigger(name = "request", methods = {HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) final String message
  * ) {
  *     return "{ \"id\": \"" + System.currentTimeMillis() + "\", \"description\": \"" + message + "\" }";
- * }</pre>
+ * }
+ * </pre>
  * 
  * @since 1.0.0
  */
 @Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.PARAMETER, ElementType.METHOD})
+@Target({ ElementType.PARAMETER, ElementType.METHOD })
 public @interface CosmosDBOutput {
     /**
      * The variable name used in function.json.
+     * 
      * @return The variable name used in function.json.
      */
     String name();
 
     /**
-     * <p>Defines how Functions runtime should treat the parameter value. Possible values are:</p>
+     * <p>
+     * Defines how Functions runtime should treat the parameter value. Possible
+     * values are:
+     * </p>
      * <ul>
-     *     <li>"" or string: treat it as a string whose value is serialized from the parameter</li>
-     *     <li>binary: treat it as a binary data whose value comes from for example OutputBinding&lt;byte[]&gt;</li>
+     * <li>"" or string: treat it as a string whose value is serialized from the
+     * parameter</li>
+     * <li>binary: treat it as a binary data whose value comes from for example
+     * OutputBinding&lt;byte[]&gt;</li>
      * </ul>
+     * 
      * @return The dataType which will be used by the Functions runtime.
      */
     String dataType() default "";
 
     /**
      * Defines the database name of the CosmosDB to which to write.
+     * 
      * @return The database name string.
      */
     String databaseName();
 
     /**
      * Defines the collection name of the CosmosDB to which to write.
+     * 
      * @return The collection name string.
      */
     String collectionName();
 
     /**
      * Defines the ID of the CosmosDB to which to write.
+     * 
      * @return The ID string.
      */
     boolean createIfNotExists() default false;
 
     /**
      * Defines the app setting name that contains the CosmosDB connection string.
+     * 
      * @return The app setting name of the connection string.
      */
     String connectionStringSetting();
+
+    /**
+     * Defines the partition key path for the created collection when
+     * createIfNotExists is set to true. May include binding parameters.
+     * 
+     * @return partitionKey of the created collection.
+     */
+    String partitionKey();
+
+    /**
+     * If CreateIfNotExists is true, defines the throughput of the created
+     * collection.
+     * 
+     * @return Throughput of the created collection.
+     */
+    int collectionThroughput();
+
+    /**
+     * Enable to use with Multi Master accounts.
+     * 
+     * @return whether to Multi Master accounts
+     */
+    boolean useMultipleWriteLocations() default false;
```

Comment:
Should this be added to Trigger and Input binding?

Reply:
Looping in @ealsur to validate, but I don't believe so. This is only for writing documents.